### PR TITLE
feat(BUX-206): unify logs

### PR DESCRIPTION
**Pull request by pawellewandowski98**
# Description

Unify logs, use zerlog

# Pull Request Checklist

- [ ] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/go-paymail/blob/main/.github/CODE_STANDARDS.md)
- [ ] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/go-paymail/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] 🏠 I tested my changes locally.
- [ ] ✅ I have provided tests for my changes.
- [ ] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [ ] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

File: server/verify.go [hunk 1]
```diff
@@ -52,12 +51,13 @@ func (c *Configuration) verifyPubKey(w http.ResponseWriter, req *http.Request, p
...
-		return
-	}
+	//w.Header().Set("Content-Type", "application/json")
+	//w.WriteHeader(http.StatusOK)
+	//err = json.NewEncoder(w).Encode(verPayload)
+	//
+	//if err != nil {
+	//	ErrorResponse(w, req, ErrorEncodingResponse, err.Error(), http.StatusBadRequest, c.Logger)
+	//	return
+	//}
```
Comment (Nazarii-4chain) [inline server/verify.go:61]: Please remove comment

**Review by Nazarii-4chain — DISMISSED** [review]

File: server/verify.go (hunk 1 above)
Comment (pawellewandowski98) [reply server/verify.go]: Done

File: examples/client/check_ssl/check_ssl.go [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
 
 import (
-	"log"
-
 	"github.com/bitcoin-sv/go-paymail"
+	"github.com/bitcoin-sv/go-paymail/logging"
 )
 
 func main() {
+	logger := logging.GetDefaultLogger()
```
Comment (dorzepowski) [inline examples/client/check_ssl/check_ssl.go:9]: It is only example,
I'm not sure if we should use logger here.

File: examples/client/convert_handle/convert_handle.go [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
 
 import (
-	"log"
-
 	"github.com/bitcoin-sv/go-paymail"
+	"github.com/bitcoin-sv/go-paymail/logging"
 )
 
 func main() {
+	logger := logging.GetDefaultLogger()
```
Comment (dorzepowski) [inline examples/client/convert_handle/convert_handle.go:9]: It is only example,
I'm not sure if we should use logger here.

File: go.mod [hunk 4]
```diff
@@ -12,24 +12,30 @@ require (
...
+	golang.org/x/sys v0.13.0 // indirect
 	golang.org/x/text v0.13.0 // indirect
 	golang.org/x/tools v0.12.0 // indirect
 	google.golang.org/genproto/googleapis/rpc v0.0.0-20230815205213-6bfd019c3878 // indirect
 	google.golang.org/grpc v1.57.0 // indirect
 	google.golang.org/protobuf v1.31.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
+
+//replace github.com/BuxOrg/bux => ../../BuxOrg/bux
```
Comment (dorzepowski) [inline go.mod:41]: remove

File: spv/spv_test.go [hunk 5]
```diff
@@ -56,7 +56,6 @@ func TestExecuteSimplifiedPaymentVerification_CorruptedTransaction_ReturnError(t
...
...9ffffffff0264000000000000001976a914ddfbaa2cd75b86cf24135603b5ddd2b1968bc62d88ac29010000000000001976a9148360b65a14e83a68aef2fae3f9cf472a90dd4da488ac000000000001000000032a796907de322523a34e801e304ea43b6938cfa5afe3ff4fea7fbc34bd52164c010000006a47304402203b0ef0d3ccba1cc3dcb9dbccdc49a7eca5a7f887564960050110a7a979cd8026022045eac010d965623c002a06946c9e9fe2a17ca0504d5eb90b67bc73a8e8904c624121033733254e143eba5ee6989fe898647d4e6170c0d82faa72fbce1c1e701960edb9ffffffff73bd2e109d1e01d9c5828e15b7e3212e54dd3d9558ceae6514c97e07ad275110000000006b483045022100b05c6873b1161dc0ea574cfbb2fdb3843782f2f7ea3c136a7086b69df8a8e5ed022019c3d8cb94c75c4c9a6acc44c9c7809a99e5cd6427b1ab5fcdd6a7dc1527331b4121033733254e143eba5ee6989fe898647d4e6170c0d82faa72fbce1c1e701960edb9ffffffff95e93ce1a376cfd842ffbd849f0ee37330d6b62a51cddfacfbed9c1a4d49cce7000000006a473044022044c557f9fe6c42e3c22e9ee898d5ae2adf11f08bdc2cee6d06954238c60b95a1022072a5b9de839bf851958b6b2aa7e8c28c43f8275333e0cf6cc272d75ec4740e7c41210275aaa422bc13c303b70026b3fd1fff37c199e5e98bd0a429922217f9b861870effffffff0220030000000000001976a914ddfbaa2cd75b86cf24135603b5ddd2b1968bc62d88ac92000000000000001976a9148360b65a14e83a68aef2fae3f9cf472a90dd4da488ac0000000000"
```
Comment (dorzepowski) [inline spv/spv_test.go:59]: Why?

File: server/utilitis.go [hunk 6]
```diff
@@ -2,31 +2,34 @@ package server
...
-	writeRespone(w, http.StatusOK, "application/json", jsonData)
+	writeResponse(w, req, log, http.StatusOK, "application/json", jsonData)
 }
 
-func writeRespone(w http.ResponseWriter, statusCode int, contentType string, responseData []byte) {
+func writeResponse(w http.ResponseWriter, req *http.Request, log *zerolog.Logger, statusCode int, contentType string, responseData []byte) {
 	w.Header().Set("Content-Type", contentType)
 	w.WriteHeader(statusCode)
 
+	log.Debug().Msgf("%d | %s | %s | %s ", statusCode, req.RemoteAddr, req.Method, req.URL)
```
Comment (dorzepowski) [inline server/utilitis.go:27]: Additionally we could add those params in contextual fields and maybe some additional marker like `Str("logger", "http-response")`

File: server/error.go [hunk 7]
```diff
@@ -55,14 +57,21 @@ var (
...
+func ErrorResponse(w http.ResponseWriter, req *http.Request, code, message string, statusCode int, log *zerolog.Logger) {
+	if log == nil {
+		log = logging.GetDefaultLogger()
+	}
+
 	srvErr := &paymail.ServerError{Code: code, Message: message}
 	jsonData, err := json.Marshal(srvErr)
 
 	if err != nil {
+		log.Debug().Msgf("%d | %s | %s | %s | %s", http.StatusInternalServerError, req.RemoteAddr, req.Method, req.URL, message)
```
Comment (dorzepowski) [inline server/error.go:69]: Additionally we could add those params in contextual fields and maybe some additional marker like Str("logger", "http-response")

File: server/error.go [hunk 8]
```diff
@@ -55,14 +57,21 @@ var (
...
 		http.Error(w, ErrorFailedMarshalJSON, http.StatusInternalServerError)
 		return
 	}
 
-	writeRespone(w, statusCode, "application/json", jsonData)
+	errorLogger := log.With().Str("code", code).Str("msg", message).Logger()
```
Comment (dorzepowski) [inline server/error.go:74]: It would be good to have some additional marker like: `Str("logger", "http-error")` or something

**Review by dorzepowski — CHANGES_REQUESTED** [review]

File: spv/spv_test.go (hunk 5 above)
Comment (pawellewandowski98) [reply spv/spv_test.go:59]: It was unused

File: examples/client/check_ssl/check_ssl.go (hunk 2 above)
Comment (pawellewandowski98) [reply examples/client/check_ssl/check_ssl.go]: Done

File: examples/client/convert_handle/convert_handle.go (hunk 3 above)
Comment (pawellewandowski98) [reply examples/client/convert_handle/convert_handle.go]: Done

File: server/utilitis.go (hunk 6 above)
Comment (pawellewandowski98) [reply server/utilitis.go]: Done

File: server/error.go (hunk 7 above)
Comment (pawellewandowski98) [reply server/error.go]: Done

File: server/error.go (hunk 8 above)
Comment (pawellewandowski98) [reply server/error.go]: Done

File: go.mod (hunk 4 above)
Comment (pawellewandowski98) [reply go.mod]: Done

**Review by dorzepowski — APPROVED** [review]

**Pull request merged**